File: DSA With SUPREME 2.0 Batch/Array/min_max.cpp
#include <iostream>
#include<limits.h>
using namespace std;
int min(int arr[], int size){
  int min = INT_MAX;
  for(int i = 0 ; i < size ; i++){
    if(min>arr[i]) min = arr[i];
  }
  return min;
}
int max(int arr[], int size){
  int max = INT_MIN;
  for(int i = 0 ; i < size ; i++){
    if(max<arr[i]) max = arr[i];
  }
  return max;
}
int main() {
  int arr[7];
  cout<<"Enter elements : ";
  for(int i = 0 ; i < 7 ; i++){
    cin>>arr[i];
  }
  cout<<"Minimum element in array : "<<min(arr,7)<<endl;
  cout<<"Minimum element in array : "<<max(arr,7);
  return 0;
}

File: DSA With SUPREME 2.0 Batch/01Array/MinMax.java
class MinMax{

    public static int[] minMax(int[] arr) {

        // Edge Case: Check if the array is null or empty
        if (arr == null || arr.length == 0) {
            throw new IllegalArgumentException("Array must not be null or empty");
        }
    
        // Initialize min and max with extreme values
        int min = Integer.MAX_VALUE;
        int max = Integer.MIN_VALUE;
    
        for (int elem : arr) {
            if (elem < min) {
                min = elem;  // Update min if current element is smaller
            }
            if (elem > max) {
                max = elem;  // Update max if current element is larger
            }
        }
    
        return new int[]{min, max};
    }    

    public static void main(String[] args){
        int[] arr = new int[]{100, -2, 30, 44, -2, 100};
        int[] result = minMax(arr);
        System.out.println("Minimum number: " + result[0]);
        System.out.println("Maximum number: " + result[1]);
    }

}